src/app/static/src/app/components/projects/ProjectHistory.vue

```diff
@@ -355,7 +487,7 @@
                 }
             },
             async confirmPromotion(name) {
-                if (this.versionToPromote) {
+                if (this.versionToPromote && this.newProjectName) {
```

Comment:
What's this? An opportunistic bug fix, not related to this ticket? Should there be an associated test?

Again, would be good to just do a quick review of your code before submitting for PR and mentioning out of the ordinary bits like like to circumvent daft questions like this one! ;)

Reply:
I thought this was obvious, when a confirm promotion button is triggered without any input entry for project name, the endpoint action gets called. A bug that is now fixed.

Reply:
> I thought this was obvious. When a confirm promotion button is triggered without any input entry for project name, the endpoint action gets called. A bug that is now fixed.

Obvious to you, perhaps, @carbowizzy since you've been working on this piece of code, but not necessarily to a reviewer who is having to do a context-switch from whatever they've had their head in. It's probably a good rule of thumb to always comment on any change in a PR which is not directly related to the ticket, no matter how trivial it appears to you.

Reply:
Sure, no validation was triggered when invoking carriage return on the input of promoteVersion modal and even rename modal has no validation. I will probably look into fixing more on maintenance day. Couldn't get a test to fail or pass for this behaviour, this is where selenium will come handy.

Reply:
Not clear what the plan is here - are you saying this is a big enough piece of work to merit a new ticket? If so then please create it, clearly explaining which modals need to be updated and why, and add a link here. Just to avoid it slipping through the cracks.

Reply:
@carbowizzy Not clear of the outcome here: was expecting to see either a test here, or a via a new ticket linked from here

Reply:
Test was added on L658 of projectHistory.

Reply:
Aha thanks very much. I didn't spot the relevance first time round! Looks good.